src/rasterstats/main.py

```diff
@@ -164,9 +242,14 @@ def gen_zonal_stats(
 
             # Mask the source data array
             # mask everything that is not a valid value or not within our geom
-            masked = np.ma.MaskedArray(
-                fsrc.array,
-                mask=(isnodata | ~rv_array))
+            if percent_cover_selection is not None:
+                masked = np.ma.MaskedArray(
+                    fsrc.array,
+                    mask=(isnodata | ~rv_array | percent_cover > percent_cover_selection))
+            else:
+                masked = np.ma.MaskedArray(
+                    fsrc.array,
+                    mask=(isnodata | ~rv_array))
```

Comment:
This will raise an `TypeError` is `rv_array` comes from `rasterize_pctcover_geom` - it won't be boolean, but instead have dtype `float32`, raising `TypeError: ufunc 'invert' not supported for the input types`.